### PR TITLE
Bomberman

### PR DESCRIPTION
**Pull request by Jaewongtongsoup**
Bomberman implementation

### CODE REVIEW

File: Retro_ML_TEST/Game/SuperBomberman3/DataFetcherTest.cs [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+            {
+                for (int j = 0; j < width; j++)
+                {
+                    output[i, j] = input[i * width + j];
+                }
+            }
+
+            return output;
+        }
+    }
```
Comment (Tourmi) [inline Retro_ML_TEST/Game/SuperBomberman3/DataFetcherTest.cs:648]: MathUtils has the exact same function, please use that one instead.

File: Retro_ML_TEST/Retro_ML_TEST.csproj [hunk 2]
```diff
@@ -28,6 +28,7 @@
     <ProjectReference Include="..\Games\Retro_ML.Tetris\Retro_ML.Tetris.csproj" />
     <ProjectReference Include="..\Games\Retro_ML.SuperMarioBros\Retro_ML.SuperMarioBros.csproj" />
     <ProjectReference Include="..\Games\Retro_ML.Metroid\Retro_ML.Metroid.csproj" />
+	<ProjectReference Include="..\Games\Retro_ML.SuperBomberman3\Retro_ML.SuperBomberman3.csproj" />
     <ProjectReference Include="..\Retro_ML\Retro_ML.csproj" />
```
Comment (Tourmi) [inline Retro_ML_TEST/Retro_ML_TEST.csproj:32]:
```suggestion
    <ProjectReference Include="..\Games\Retro_ML.Metroid\Retro_ML.Metroid.csproj" />
    <ProjectReference Include="..\Games\Retro_ML.SuperBomberman3\Retro_ML.SuperBomberman3.csproj" />
    <ProjectReference Include="..\Retro_ML\Retro_ML.csproj" />
```

**Review by Tourmi — CHANGES_REQUESTED** [review]

File: Games/Retro_ML.SuperBomberman3/Game/SB3DataFetcher.cs [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+                        }
+                    }
+                }
+            }
+
+            return new Tuple<double, double>(closestX, closestY);
+        }
+
+        /// <summary>
+        /// Useful to keep track of the bombs that are being planted on the map but needs to be called for every frames.
```
[the remark this answers is not in the document]
Comment (Jaewongtongsoup) [reply Games/Retro_ML.SuperBomberman3/Game/SB3DataFetcher.cs:358]: Resolved

File: Retro-ML.sln [hunk 4]
```diff
@@ -12,10 +12,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Retro_ML.Application", "App
 		{6CD071EB-0A85-48F0-8562-117F5DB475EE} = {6CD071EB-0A85-48F0-8562-117F5DB475EE}
 		{7735343C-8DCF-4061-B168-0CAA36AA4159} = {7735343C-8DCF-4061-B168-0CAA36AA4159}
 		{78B90532-499C-4803-8595-CBC5C35F74DE} = {78B90532-499C-4803-8595-CBC5C35F74DE}
-		{9F3C34AD-2E4C-47DC-9715-D1CB9C6B8993} = {9F3C34AD-2E4C-47DC-9715-D1CB9C6B8993}
-		{B7EAE7F6-D1D9-492D-BE35-15F018AADC17} = {B7EAE7F6-D1D9-492D-BE35-15F018AADC17}
-		{CAAF91C4-AF31-4682-910B-FF765ED44BBF} = {CAAF91C4-AF31-4682-910B-FF765ED44BBF}
 		{F5E1C73C-1ECF-4B27-8E6F-A4AEC6E9C29B} = {F5E1C73C-1ECF-4B27-8E6F-A4AEC6E9C29B}
+		{9F3C34AD-2E4C-47DC-9715-D1CB9C6B8993} = {9F3C34AD-2E4C-47DC-9715-D1CB9C6B8993}
 	EndProjectSection
```
[the remark this answers is not in the document]
Comment (Jaewongtongsoup) [reply Retro-ML.sln:17]: Resolved

**Review by Tourmi — APPROVED** [review]

**Pull request merged**